url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/com/wroblicky/andrew/joust/game/GameSetup.java

```diff
@@ -0,0 +1,505 @@
+		Location c3 = new Location("c3");
+		board[2][2] = c3;
+		Location d3 = new Location("d3");
+		board[2][3] = d3;
+		Location e3 = new Location("e3");
+		board[2][4] = e3;
+		Location f3 = new Location("f3");
+		board[2][5] = f3;
+		Location g3 = new Location("g3");
+		board[2][6] = g3;
+		Location h3 = new Location("h3");
+		board[2][7] = h3;
+		
+		Location a4 = new Location("a4");
+		board[3][0] = a4;
+		Location b4 = new Location("b4");
+		board[3][1] = b4;
+		Location c4 = new Location("c4");
+		board[3][2] = c4;
+		Location d4 = new Location("d4");
+		board[3][3] = d4;
+		Location e4 = new Location("e4");
+		board[3][4] = e4;
+		Location f4 = new Location("f4");
+		board[3][5] = f4;
+		Location g4 = new Location("g4");
+		board[3][6] = g4;
+		Location h4 = new Location("h4");
+		board[3][7] = h4;
+		
+		Location a5 = new Location("a5");
+		board[4][0] = a5;
+		Location b5 = new Location("b5");
+		board[4][1] = b5;
+		Location c5 = new Location("c5");
+		board[4][2] = c5;
+		Location d5 = new Location("d5");
+		board[4][3] = d5;
+		Location e5 = new Location("e5");
+		board[4][4] = e5;
+		Location f5 = new Location("f5");
+		board[4][5] = f5;
+		Location g5 = new Location("g5");
+		board[4][6] = g5;
+		Location h5 = new Location("h5");
+		board[4][7] = h5;
+		
+		Location a6 = new Location("a6");
+		board[5][0] = a6;
+		Location b6 = new Location("b6");
+		board[5][1] = b6;
+		Location c6 = new Location("c6");
+		board[5][2] = c6;
+		Location d6 = new Location("d6");
+		board[5][3] = d6;
+		Location e6 = new Location("e6");
+		board[5][4] = e6;
+		Location f6 = new Location("f6");
+		board[5][5] = f6;
+		Location g6 = new Location("g6");
+		board[5][6] = g6;
+		Location h6 = new Location("h6");
+		board[5][7] = h6;
+		
+		Location a7 = new Location("a7");
+		board[6][0] = a7;
+		Location b7 = new Location("b7");
+		board[6][1] = b7;
+		Location c7 = new Location("c7");
+		board[6][2] = c7;
+		Location d7 = new Location("d7");
+		board[6][3] = d7;
+		Location e7 = new Location("e7");
+		board[6][4] = e7;
+		Location f7 = new Location("f7");
+		board[6][5] = f7;
+		Location g7 = new Location("g7");
+		board[6][6] = g7;
+		Location h7 = new Location("h7");
+		board[6][7] = h7;
+		
+		Location a8 = new Location("a8");
+		board[7][0] = a8;
+		Location b8 = new Location("b8");
+		board[7][1] = b8;
+		Location c8 = new Location("c8");
+		board[7][2] = c8;
+		Location d8 = new Location("d8");
+		board[7][3] = d8;
+		Location e8 = new Location("e8");
+		board[7][4] = e8;
+		Location f8 = new Location("f8");
+		board[7][5] = f8;
+		Location g8 = new Location("g8");
+		board[7][6] = g8;
+		Location h8 = new Location("h8");
+		board[7][7] = h8;
+		try {
+			// Open the file that is the first 
+			// command line parameter
+			FileInputStream fstream = new FileInputStream(initialConfig);
+			// Get the object of DataInputStream
+			DataInputStream in = new DataInputStream(fstream);
+			BufferedReader br = new BufferedReader(new InputStreamReader(in));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *br*. The resource is referenced at lines 343, 344, 347. The resource is closed at line 498. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).